internal/controller/validatorconfig_controller.go

```diff
@@ -100,7 +107,14 @@ func (r *ValidatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		if err := r.deletePlugins(ctx, vc); err != nil {
 			return ctrl.Result{}, err
 		}
-		return ctrl.Result{}, removeFinalizer(ctx, r.Client, vc, CleanupFinalizer)
+
+		err = removeFinalizer(ctx, r.Client, vc, CleanupFinalizer)
+
+		if emitErr := r.emitFinalizeCleanup(); err != nil {
+			r.Log.Error(emitErr, "Failed to emit FinalizeCleanup request")
```

Comment:
Should this be checking `emitErr != nil` instead of `err != nil`? (or both?)

Reply:
good catch!